### DISCUSSION TITLE
Support GtkSourceView

### DISCUSSION THREAD

**Discussion by ghost** [geany/geany#3695] [Ideas]
Let the users choose to use the syntax highlighting engine they prefer in `Preferences`. Scintilla is still the default syntax highlighting engine. The advantage of GtkSourceView is it's much easier to add support for new languages compared to Scintilla.

**Comment by elextr** [comment]
GtkSourceView and Scintilla are two different libraries addressing the same use-case but with totally different ways of doing it, different APIs and different highlighting approaches (as you observed).  They are not drop in replacements.  Geany and plugins are entirely written around the Scintilla API, relying on Scintilla features and capabilities.  To change it to GtkSourceView would require a complete rewrite of all of those, it would not be Geany, it would be a different program.

There is nothing preventing "somebody" writing a Scintilla lexer that uses regexen or other user configurable method to generate highlighting, but nobody has found it worthwhile.
